test/mitmproxy/addons/test_savehar.py

```diff
@@ -173,6 +173,18 @@ def test_savehar(log_file: Path, tmp_path: Path, monkeypatch):
     assert actual_har == expected_har
 
 
+def test_flow_entry():
+    s = SaveHar()
+    flow = tflow.twebsocketflow(
+        method="CONNECT", host="test.test", port=443, scheme="https"
```

Comment:
A WebSocket flow shouldn"t have a CONNECT method, AFAIK that's typically GET with Upgrade headers? Or is this something you've seen in the wild?

Reply:
You are right, I think `ttcpflow` should be used instead? Correct me if I'm wrong

Reply:
This should be an HTTP flow. :)

Reply:
Due to my limited knowledge and understanding of the code-base, I looked at some other tests in `test_savehar.py` and concluded that `twebsocketflow` can be used for generating a sample flow without knowing it's actual use. My apologies for that.

As you said, no changes are required in `tflow.py`.

Instead, we can use `tflow` and pass a `CONNECT` request in order to generate a sample HTTP flow for testing.